src/Utils.js

```diff
@@ -189,7 +189,7 @@ export function selectorError(selector, type = '') {
   );
 }
 
-export const isCompoundSelector = /([a-z]\.[a-z]|[a-z]\[.*\]|[a-z]#[a-z])/i;
+export const isCompoundSelector = /^[\.#]?-?[_a-z]+[_a-z0-9-]*[\.\[#]/i;
```

Comment:
i wish we had commented regexps for situations like this. Let's see:
```js
/
  ^               # start
  [\.#]?          # anything but a hash, 0 or 1
  -?              # a hyphen, 0 or 1
  [_a-z]+         # any letter or underscore, 1 or more
  [_a-z0-9-]*     # any letter, underscore, number, or hyphen, 0 or more
  [\.\[#]         # any dot, left bracket, or hash
/i
```

Even after that I'm not really sure how to reason out what it's matching against. I wonder if we could write this as multiple different regexes, instead of trying to combine it into one case?